### ISSUE TITLE
[request] Support Background Contrast effect

### ISSUE THREAD

**Issue by Luwx** [tsujan/Kvantum#303]
Kwin has a effect that can change the color contrast and saturation of the area bellow a window. Currently it's only implemented in some plasma themes but the effect can be extended to the whole desktop, like the blur effect.

Themes that use transparency can benefit from this, as they don't need to be as transparent to get the color of what is behind them (similar to iOS).

![screenshot_20181124_233323](https://user-images.githubusercontent.com/33076480/48974542-51c2dc00-f043-11e8-86b0-6f0f82718768.png)

[Effect documentation](https://api.kde.org/frameworks/kwindowsystem/html/namespaceKWindowEffects.html#a166a5fc7121ecf57a69fd19545c6f724)

**Comment by tsujan** [comment]
Will investigate it. Thanks for the idea!

**Comment by tsujan** [comment]
After giving it some thought, I decided to use this design (in Kvantum manager):

![design](https://user-images.githubusercontent.com/981076/48976233-f28fb680-f098-11e8-8fcf-ab173533bf85.png)

As the tooltip says, the contrast effect is enabled only for windows that support the blur effect but the KDE blur effect can be disabled. In this way, the code of Kvantum won't be cluttered.

This requires another change too: A similar setting is needed in [BreezeEnhanced](https://github.com/tsujan/BreezeEnhanced) for the title-bar to match the window. I'll think about it later...

**Comment by Luwx** [comment]
Awesome!
Thank you very much, you are the best :)

**Comment by tsujan** [comment]
Thank you for making me see this use cases!

**Comment by trmdi** [comment]
Can this help to increase the darkness of the background behind a window **automatically** ?
e.g. When a transparent Konsole is on top of a light background webpage, it's harder to see the Konsole content compared to when the background behind the Konsole window is a darker color.

**Comment by tsujan** [comment]
I don't think it distinguishes between light and dark backgrounds in that way. For example, when intensity and saturation are below 1.00, translucent windows look darker against all backgrounds.

I like to set saturation to 2.00. The blur effect is enough for readability.

**Comment by tsujan** [comment]
The problem is that it cannot be implemented for BreezeEnhanced :( These settings are beyond its code.

**Comment by Luwx** [comment]
@trmdi Decreasing the contrast value and increasing the saturation can help a bit, but ideally a good way to solve this would be a brightness/lightness control in the effect..

**Comment by Luwx** [comment]
@tsujan Isn't it possible for Kvantum to also apply the blur to decorations, like this? `xprop -f _KDE_NET_WM_BLUR_BEHIND_REGION 32c -set _KDE_NET_WM_BLUR_BEHIND_REGION 0` 

The contrast effect could be applied the same way. This would also solve https://bugs.kde.org/show_bug.cgi?id=395725 since Kvantum can specify which areas to blur (no more sharp edges on rounded corners).

**Comment by tsujan** [comment]
Oh, no, it isn't possible. The decoration is beyond what QStyle can control. There isn't a single QStyle parameter about window decoration (`CC_TitleBar` and `PM_TitleBar...` parameters are about `QMdiSubWindow`).

The situation is quite different with GTK+: they integrated the decoration into their style. I prefer the Qt way because it's more flexible.

**Comment by Luwx** [comment]
Makes sence. I thought about extending the effect area into the decoration but this would be a very ugly hack.

**Comment by tsujan** [comment]
> but this would be a very ugly hack.

Actually, it's impossible. What will happen with Qt6, we should wait to know.
